File: apps/fogg/src/pump/once.go
package pump

import "fogg/src/type"

func once(address types.Address, payload types.Payload, options types.Options) int {
	if WithFn == nil {
		return -1
	}
	for _, def := range WithFn() {
		if def.Fit(address, payload, options) {
			return def.Fun(address, payload, options)
		}
	}
	return -1
}

File: apps/rata/src/pump/once.rs
use crate::defs::with::with;
use crate::r#type::{Address, Options, Payload};

pub fn once(address: Address, payload: Payload, options: Options) -> i32 {
    for def in with() {
        if (def.fit)(address, payload, options) {
            return (def.fun)(address, payload, options);
        }
    }
    -1
}
